repositories/experiment.repository.js

```diff
@@ -1,40 +1,50 @@
-const MongoStorage = require("../db/MongoStorage");
+const { MongoStorage }= require("../db/MongoStorage");
+const validateDate = require("validate-date");
 
-module.exports = class UsersRepository {
+module.exports = new class ExperimentsRepository extends  MongoStorage {
     constructor() {
-        this.storage = new MongoStorage("experiment");
+        super("experiment")
     }
 
-    find() {
-        return this.storage.find();
-    }
-
-    retrieve(id) {
-        return this.storage.retrieve(id);
+    findByDate(year, month) {
+        if (validateDate(`${month}/01/${year}`)) {
+            const start = new Date(year, month, 1);
+            const end = new Date(year, month, 31);
+            return this.Model.countDocuments({
+                end_time: {
+                    $gte: start,
+                    $lte: end
+                }
+            });
+        }
     }
 
-    create(experiment) {
-        const user = this.storage.create(experiment);
-        return user;
+    incVariantSuccessCount(id, variant){
+        return this.update(id, {$inc : {[`variant_success_count.${variant}`]: 1}});
     }
 
-    update(id, experiment) {
-        return this.storage.update(id, experiment);
+    getVariantSuccessCount(id){
+        const experiment = this.retrieve(id);
+        if(!! experiment)
+            return experiment.variant_success_count;
+        else
+            return null;
```

Comment:
instead of returning null, you can do the check if(!experiment) throw new NotFoundEntity('experiment')